src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -47,184 +47,14 @@ namespace winrt::TerminalApp::implementation
         if (keyIter != _keyShortcuts.end())
         {
             const auto actionAndArgs = keyIter->second;
-            return _DoAction(actionAndArgs);
+            return _dispatch.DoAction(actionAndArgs);
         }
         return false;
     }
 
-    bool AppKeyBindings::_DoAction(ActionAndArgs actionAndArgs)
+    void AppKeyBindings::SetDispatch(const winrt::TerminalApp::ShortcutActionDispatch& dispatch)

```

Comment:
Does this only ever happen once? If so, any reason it can't be part of construction?

Reply:
Yes it only happens once, but `GlobalAppSettings` is responsible for building the `AppKeyBindings`, while the `TerminalPage` owns the `ShortcutActionDispatch`. So we'd have to plumb the `ShortcutActionDispatch` down another level into the globals, which I don't totally love :/